app/presenters/spree/feed_product_presenter.rb

```diff
@@ -0,0 +1,274 @@
+module Spree
+  class SchemaError < StandardError
+    def initialize msg, product
+      super("Missing mandatory #{msg}. Skipping feed entry for #{product.inspect}")
+    end
+  end
+
+  class FeedProductPresenter
+    # @!attribute schema
+    #   @return [Array <Symbol, Hash>] the nested schema to use in xml generation
+    #
+    # @!attribute properties
+    #   @return [Array <Symbol>] the product properties list to use in accessor creation.
+    attr_accessor :schema, :properties
+    # Creates FeedProductPresenter for presenting products as items
+    # in RSS feed for Google Merchant
+    #
+    # @param view [ActionView view context] the view being rendered.
+    #
+    # @param product [Spree::Product] the product to display. It must
+    #   have its own landing page, which is why variants are not supported
+    #   at this time.
+    #
+    # @param properties [Array <Symbol>] all of the product data which is
+    #   obtained from the product.properties
+    def initialize(view, product, properties: nil)
+
+      @view = view
+      @product = product
+      @schema = [
+        :id, :title, :description, :image_link, :price, :availability,
+        :identifier_exists, :link, :condition,
+        {parent: :shipping, schema: [:price]},
+        {parent: :tax, schema: [:rate]},
+      ]
+
+      # Automatically include any product_property which defines
+      # values for any of the following:
+      @properties = [:brand, :gtin, :mpn, :google_product_category,
+                     :adult, :multipack, :is_bundle, :energy_efficiency_class,
+                     :age_group, :color, :gender, :material, :pattern, :size,
+                     :size_type, :size_system, :item_group_id, :product_type,
+                     :unit_pricing_measure, :unit_pricing_base_measure]
+
+      # For each property listed, if the product has a property
+      # associated with it which matches, create an instance method
+      # of the same name which retrieves the property value.
+      @properties.each do |prop|
+        next unless @product.property(prop.to_s)
+        @schema << prop
+        self.class.send(:define_method, prop) do
+          @product.property(prop.to_s)
+        end
+      end
+    end
+
+    # Creates an <item> RSS feed entry of the
+    # product, corresponding with Google's requested schema. If a
+    # mandatory element of the schema is missing, a SchemaError is
+    # raised, the entire <item> entry for this product is skipped,
+    # and an error is logged to the configured log file or STDERR.
+    #
+    # @param xml [Builder::XmlMarkup]
+    # @return String, the xml <item> tag and content for this product.
+    def item xml
```

Comment:
I think we can separate the logic for generating XML, and the logic for the helper method.